CONTRIBUTING.md

```diff
@@ -1,52 +1,122 @@
 
-## Maintainers
+[prs]: https://github.com/rust-windowing/winit/pulls
+[issues]: https://github.com/rust-windowing/winit/issues
+[matrix]: https://matrix.to/#/#rust-windowing:matrix.org
 
-The current maintainers for each platform are listed in the [CODEOWNERS](.github/CODEOWNERS) file.
+## Maintainers
 
-## Release process
+Winit has plenty of maintainers with different backgrounds, different time
+available to work on Winit, and reasons to be winit maintainer in the first
+place. To ensure that winit's code quality is not decreasing over time and to
+make it easier to teach new maintainers the _winit way of doing things_ the
+common policies and routines are defined in this section.
+
+The current maintainers for each platform are listed in [this file][CODEOWNERS].
+
+### Contributions handling
+
+The maintainers must ensure that the external contributions meet the winit's
+quality standards. If it's not, it **is the maintainer's responsibility** to
+bring it on par, which includes:
+
+  - Ensure that formatting is consistent and `CHANGELOG` messages are clear
+    for the end users.
+  - Improve the commit message, so it'll be easier for other maintainers to
+    understand the motivation without going through all the discussions on the
+    particular patch/PR.
+  - Ensure that the proposed patch doesn't break platform parity. If the
+    breakage is desired by contributor, an issue should be opened to discuss
+    with other maintainers before merging.
+  - Always fix CI issues before merging if they don't originate from the
+    submitted work.
+
+However, maintainers should give some leeway for external contributors, so they
+don't feel discouraged contributing, for example:
+
+  - Suggest a patch to resolve style issues, if it's the only issue with the
+    submitted work. Keep in mind that pushing the resolution yourself is not
+    desired, because contributor might not agree with what you did.
+  - Be more explicit on how things should be done if you doesn't like the
+    approach.
+  - Suggest to finish PR for them if they're absent for a while and you need the
+    proposed changes to move forward with something. In such case maintainer
+    must preserve attribution with `Co-authored-by`, `Suggested-by`, or keep
+    the original committer.
+  - Rebase their work for them when massive changes to winit codebase were
+    introduced.
+
+When reviewing code of other maintainers all of the above is on the maintainer
+who submitted the patch. Interested maintainers could help push the work over
+the finish line, but teaching other maintainers should be preferred.
+
+For a _regular_ contributor to winit, the maintainer should slowly start
+requiring contributor to match *maintainer* quality standards when writing
+patches and commit messages.
+
+### Contributing
+
+When submitting a patch maintainer should follow the general contributing
+guidelines, however all soft rules (e.g `Try to`), become strict.
```

Comment:
Hmm, what are the soft rules that this refers to?

Reply:
I think those soft rules were removed in this fork of the original PR